spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/client/loadbalancer/impl/context/RoundRobinLoadBalancer.java

```diff
@@ -0,0 +1,90 @@
+package org.springframework.cloud.client.loadbalancer.impl.context;
+
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.cloud.client.ServiceInstance;
+import org.springframework.cloud.client.loadbalancer.impl.scoped.ScopedDiscoveryClient;
+import org.springframework.cloud.client.loadbalancer.impl.support.LoadBalancerClientFactory;
+
+import static org.springframework.util.CollectionUtils.isEmpty;
+
+/**
+ * @author Spencer Gibb
+ */
+public class RoundRobinLoadBalancer implements LoadBalancer<ServiceInstance> {
```

Comment:
More of an observation than anything else.  This round robin implementation is dependent on the list of services returned from the discovery client.  There is no guarantee that the order of that list will be the same.  So in some cases we might not always be following a round robin algorithm.  Specifically once the service list is refreshed from the service discovery service we cant guarantee that when we go to select the next server we wont select the previous server we just tried.  This happens with Ribbon today as well.  Like I said just a comment on the way this works more than anything else.